transport/wsn/src/main/java/org/kaazing/gateway/transport/wsn/WsnAcceptor.java

```diff
@@ -1092,6 +1093,14 @@ private void doUpgrade(final HttpAcceptSession session) throws Exception {
 
                     final ResourceAddress wsLocalAddress = localAddress;
 
+                    ScheduledExecutorService taskExecutor = Executors.newScheduledThreadPool(1);
+
+                    Long handshakeTimeout = 10000L;//localAddress.getOption(HANDSHAKE_TIMEOUT).longValue();
+                    if (handshakeTimeout != null && handshakeTimeout > 0) {
+                        session.getFilterChain().addLast("wsnStartHandshakeTimerFilter",
+                                new WsxStartHandshakeTimerFilter(logger, handshakeTimeout, taskExecutor));
```

Comment:
Same here, don't create the taskExecutor, you can then pass the scheduled task around as a session key on the WsxSession